### PR TITLE
grass planting sans

### PR DESCRIPTION
**Pull request by SansPapyrus683**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

**Comment by SansPapyrus683** [comment]
still need to add comments for solution 2

File: solutions/usaco-894.mdx [hunk 1]
```diff
@@ -19,222 +19,266 @@ It can be shown that the tree can also be colored accordingly in
...
+</PySection>
 </LanguageSection>
 
 ## Solution 2 - DFS
 
 **Time Complexity**: $\mathcal{O}(N)$
 
-This can also be solved using DFS. The solution is less elegant but generates a
-valid assignment of grass types in the $\texttt{ans}$ array.
+This can also be solved using [DFS](https://usaco.guide/silver/dfs).
```
Comment (ryanchou-dev) [inline solutions/usaco-894.mdx:120]:
```suggestion
This can also be solved using [DFS](/silver/dfs).
```

File: solutions/usaco-894.mdx [hunk 2]
```diff
@@ -19,222 +19,266 @@ It can be shown that the tree can also be colored accordingly in
...
+	neighbors = [[] for _ in range(field_num)]
+	for _ in range(field_num - 1):
+		field1, field2 = [int(i) - 1 for i in read.readline().split()]
+		neighbors[field1].append(field2)
+		neighbors[field2].append(field1)
+
+needed_types = [0 for _ in range(field_num)]
+needed_types[0] = 1
+
+stack = [(0, 0)]
```
Comment (ryanchou-dev) [inline solutions/usaco-894.mdx:265]: i think `todo` might be a better name (completely up to you)

**Review by ryanchou-dev — APPROVED** [review]
so much better

File: solutions/usaco-894.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply solutions/usaco-894.mdx:265]: makes sense, will do

**Comment by SansPapyrus683** [comment]
@ryanchou-dev

File: solutions/usaco-894.mdx [hunk 3]
```diff
@@ -19,222 +19,292 @@ It can be shown that the tree can also be colored accordingly in
...
 </LanguageSection>
 
 ## Solution 2 - DFS
 
 **Time Complexity**: $\mathcal{O}(N)$
 
-This can also be solved using DFS. The solution is less elegant but generates a
-valid assignment of grass types in the $\texttt{ans}$ array.
+This can also be solved using [DFS](/silver/dfs).
+Although it is less elegant, it gets the job done.
```
Comment (ryanchou-dev) [inline solutions/usaco-894.mdx:121]: i think the advantage for this is that you've gotten a coloring for all of the fields

**Review by ryanchou-dev — APPROVED** [review]
looks good

File: solutions/usaco-894.mdx (hunk 3 above)
Comment (SansPapyrus683) [reply solutions/usaco-894.mdx:121]: tru

**Review by TruaShamu — APPROVED** [review]
looks good to me

**Review by dongliu0426 — APPROVED** [review]

**Pull request merged**